validator/src/main.rs

```diff
@@ -13,131 +13,108 @@ fn main() {
     let gil = Python::acquire_gil();
     let py = &mut gil.python();
 
+    macro_rules! py_unwrap {
+        ($expr:expr) => {
+            $expr.map_err(|err| err.print(*py)).unwrap()
+        }
+    }
```

Comment:
Prefer generic functions to macros when possible:
```
fn py_unwrap<T>(py: Python, result: PyResult<T>) -> T {
  result.map_err(|err| err.print(py)).unwrap()
}
```

The above macro makes assumptions about the names of variables outside its scope (`py` in this case) which is not a good pattern.